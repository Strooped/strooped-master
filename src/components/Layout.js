import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './Layout.scss';

// eslint-disable-next-line no-unused-vars
const Layout = (props) => {
  const {
    children,
    type = null,
    hasExitButton = false,
  } = props;

  return <div className="genericlayout">
    <header className="navbar">
      <span className="title is-3 has-text-light">Strooped</span>
      {hasExitButton && <a href="/" className="button is-danger exitbutton">Exit game</a>}
    </header>
    <main className={classNames('pagecontent', type ? `is-${type}` : null)}>
      {children}
    </main>
  </div>;
};

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  // pageTitle should be used to update the tab name
  pageTitle: PropTypes.node.isRequired,
  type: PropTypes.oneOf(['centered']),
  hasExitButton: PropTypes.bool,
};

export default Layout;
