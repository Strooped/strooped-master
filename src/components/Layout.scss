@import '../styles/variables';

.genericlayout {
  background-color: $primary-background;

  min-height: 100vh;
  margin: 0;

  padding: 0;
}

.navbar {
  align-items: center;
  background-color: transparent;

  display: flex;

  justify-content: space-between;

  padding: 1em;

  > .title {
    margin-bottom: 0;
  }

  > .exitbutton {
    margin-left: auto;
  }
}

.pagecontent {
  box-sizing: border-box;
  padding: 0.5rem 0;

  &.is-centered {
    max-width: 50rem;
    margin: 0 auto;

    padding-left: 0.5rem;
    padding-right: 0.5rem;
  }
}
