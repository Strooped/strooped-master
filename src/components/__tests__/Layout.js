import React from 'react';
import { render } from 'enzyme';

import Layout from '../Layout';

describe('<Layout/>', () => {
  it('renders with children wrapped inside', () => {
    const wrapper = render(<Layout><h1>Hello world</h1></Layout>);

    expect(wrapper).toMatchSnapshot();
  });
});
