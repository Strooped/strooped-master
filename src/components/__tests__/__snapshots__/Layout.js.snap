// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout/> renders with children wrapped inside 1`] = `
<div
  class="genericlayout"
>
  <header
    class="navbar"
  >
    <span
      class="title is-3 has-text-light"
    >
      Strooped
    </span>
  </header>
  <main
    class="pagecontent"
  >
    <h1>
      Hello world
    </h1>
  </main>
</div>
`;
